File: frontend/src/types/site.ts
import type { SiteGeometry } from './common'
import type { DisturbanceRisk, EcosystemType, MonitoringStatus, ProjectStatus } from './enums'

export interface SiteLatestSnapshot {
  recorded_at: string
  carbon_captured_tco2e: number
  biodiversity_score: number
  species_count: number
  ecosystem_health_score: number
  disturbance_risk: DisturbanceRisk
}

export interface SiteProjectSummary {
  id: string
  name: string
  status: ProjectStatus
  color: string
}

export interface Site {
  id: string
  project_id: string
  name: string
  site_code: string
  ecosystem_type: EcosystemType
  monitoring_status: MonitoringStatus
  notes: string | null
  boundary: SiteGeometry
  area_hectares: number
  created_at: string
  updated_at: string
  latest_snapshot: SiteLatestSnapshot | null
  /** Only present on GET /sites/{id} per the contract. */
  project?: SiteProjectSummary
}

export interface SiteListResponse {
  items: Site[]
  total: number
}

export interface SiteUpsertRequest {
  name: string
  site_code: string
  ecosystem_type: EcosystemType
  monitoring_status: MonitoringStatus
  boundary: SiteGeometry
  notes: string | null
}

File: backend/app/schemas/site.py
import uuid
from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, Field

from app.models.enums import DisturbanceRisk, EcosystemType, MonitoringStatus, ProjectStatus


class SiteCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    site_code: str = Field(min_length=1, max_length=64)
    ecosystem_type: EcosystemType
    monitoring_status: MonitoringStatus
    boundary: dict[str, Any]
    notes: str | None = None


class SiteUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    site_code: str | None = Field(default=None, min_length=1, max_length=64)
    ecosystem_type: EcosystemType | None = None
    monitoring_status: MonitoringStatus | None = None
    boundary: dict[str, Any] | None = None
    notes: str | None = None


class LatestSnapshot(BaseModel):
    recorded_at: date
    carbon_captured_tco2e: float
    biodiversity_score: float
    species_count: int
    ecosystem_health_score: float
    disturbance_risk: DisturbanceRisk


class ProjectSummary(BaseModel):
    id: uuid.UUID
    name: str
    status: ProjectStatus
    color: str


class SiteRead(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    name: str
    site_code: str
    ecosystem_type: EcosystemType
    monitoring_status: MonitoringStatus
    notes: str | None
    boundary: dict[str, Any]
    area_hectares: float
    created_at: datetime
    updated_at: datetime
    latest_snapshot: LatestSnapshot | None = None
    project: ProjectSummary | None = None


class SiteListResponse(BaseModel):
    items: list[SiteRead]
    total: int
